### PR TITLE
Pass request header to callbacks

### PR DESCRIPTION
**Pull request by esteve**
Connects to #21

### CODE REVIEW

**Comment by esteve** [comment]
@dirk-thomas @tfoote @wjwwood

**Comment by dirk-thomas** [comment]
Do we only expose the signature which includes the header or also have a separate signature without - in case the user does not care about the information?

**Comment by tfoote** [comment]
+1

File: rclcpp/include/rclcpp/service.hpp [hunk 1]
```diff
@@ -88,13 +88,26 @@ class Service : public ServiceBase
   typedef std::function<
       void (const std::shared_ptr<typename ServiceT::Request> &,
       std::shared_ptr<typename ServiceT::Response> &)> CallbackType;
+
+  typedef std::function<
+      void (const std::shared_ptr<rmw_request_id_t> &,
+      const std::shared_ptr<typename ServiceT::Request> &,
+      std::shared_ptr<typename ServiceT::Response> &)> CallbackWithHeaderType;
```
Comment (dirk-thomas) [inline rclcpp/include/rclcpp/service.hpp]: The wrapping looks a bit unnatural. Would it be better to add a newline after `void(` and indent all arguments one level?

**Comment by esteve** [comment]
@dirk-thomas @wjwwood ping

**Comment by wjwwood** [comment]
lgtm

**Comment by dirk-thomas** [comment]
lgtm

**Comment by dirk-thomas** [comment]
We might want to squash changes like this in the future to keep browsing the history of changes easier.

**Pull request merged**